scribus/plugins/import/pdf/slaoutput.h

```diff
@@ -153,6 +153,95 @@ class AnoOutputDev : public OutputDev
 	QStringList *m_importedColors;
 };
 
+/* PDF TextBox Framework */
+/*
+* Holds all the dtails for each glyph in the text imported from the pdf file.
+*
+*/
+struct PdfGlyph
+{
+	double dx;  // X advance value
+	double dy;  // Y advance value
+	double rise;    // Text rise parameter
+	QChar code;   // UTF-16 coded character
+};
+
+
+class TextRegionLine
+{
+public:
+	qreal maxHeight = 0;
+	//we can probably use maxHeight for this.
+	qreal width = 0;
+	int glyphIndex = 0;
+	QPointF baseOrigin = QPointF(0, 0);
+	std::vector<TextRegionLine> segments = std::vector<TextRegionLine>();
+
+};
+
+class TextRegion
+{
+public:
+	enum class FrameworkLineTests
+	{
+		FIRSTPOINT,
+		SAMELINE,
+		STYLESUPERSCRIPT,
+		STYLENORMALRETURN,
+		STYLEBELOWBASELINE,
+		NEWLINE,
+		ENDOFLINE, //TODO: Implement an end of line test
+		FAIL
+	};
+
+	QPointF textRegioBasenOrigin = QPointF(0, 0);
+	qreal maxHeight = 0;
+	qreal lineSpacing = 1;
+	std::vector<TextRegionLine> textRegionLines = std::vector<TextRegionLine>();
+	qreal maxWidth = 0;
+	QPointF lineBaseXY = QPointF(0, 0); //updated with the best match left value from all the textRegionLines and the best bottom value from the textRegionLines.segments;
+	QPointF lastXY = QPointF(0, 0);
+	static bool coLinera(qreal a, qreal b);
+	bool closeToX(qreal x1, qreal x2);
+	bool closeToY(qreal y1, qreal y2);
+	bool adjunctLesser(qreal testY, qreal lastY, qreal baseY);
+	bool adjunctGreater(qreal testY, qreal lastY, qreal baseY);
+	TextRegion::FrameworkLineTests linearTest(QPointF point, bool xInLimits, bool yInLimits);
+	TextRegion::FrameworkLineTests isRegionConcurrent(QPointF newPoint);
+	TextRegion::FrameworkLineTests moveToPoint(QPointF newPoint);
+	TextRegion::FrameworkLineTests addGlyphAtPoint(QPointF newGlyphPoint, PdfGlyph new_glyph);
+	void renderToTextFrame(PageItem* textNode, ParagraphStyle& pStyle);
+	std::vector<PdfGlyph> glyphs;
+	bool isNew();
+};
+
+class AddCharInterface
+{
+public:
+	// Pure Virtual Function
+	virtual void addChar(GfxState* state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, Unicode const* u, int uLen) = 0;

```

Comment:
i wonder if something like this would help:

```cpp
struct CharAndPosition
{
    double x;
    double y;
    double dx;
    double dy;
    double originX;
    double originY;
    CharCode code;
    int nBytes;
    Unicode const* u;
    int uLen;
};
```

the name of the struct is bad.

Reply:
I think it's overkill populating a struct just for a few lines of code